tests/test_sdk.py

```diff
@@ -1069,3 +1072,351 @@ def check_project_jobs(self, project, mock_post):
+        response_mock(mock_get, [])
+        server = mindsdb_sdk.connect()
+        assert len(server.agents.list()) == 0
+
+        created_at = dt.datetime(2000, 3, 1, 9, 30)
+        updated_at = dt.datetime(2001, 3, 1, 9, 30)
+        response_mock(mock_get, [
+            {
+                'id': 1,
+                'name': 'test_agent',
+                'project_id': 1,
+                'model_name': 'test_model',
+                'skills': [],
+                'params': {},
+                'created_at': created_at,
+                'updated_at': updated_at
+            }
+        ])
+        all_agents = server.agents.list()
+        # Check API call.
+        assert mock_get.call_args.args[0] == f'{DEFAULT_LOCAL_API_URL}/api/projects/mindsdb/agents'
+
+        assert len(all_agents) == 1
+        expected_agent = Agent(
+            'test_agent',
+            'test_model',
+            [],
+            {},
+            created_at,
+            updated_at
+        )
+        assert all_agents[0] == expected_agent
+
+    @patch('requests.Session.get')
+    def test_get(self, mock_get):
+        server = mindsdb_sdk.connect()
+        created_at = dt.datetime(2000, 3, 1, 9, 30)
+        updated_at = dt.datetime(2001, 3, 1, 9, 30)
+        response_mock(mock_get,
+            {
+                'id': 1,
+                'name': 'test_agent',
+                'project_id': 1,
+                'model_name': 'test_model',
+                'skills': [],
+                'params': {},
+                'created_at': created_at,
+                'updated_at': updated_at
+            }
+        )
+        agent = server.agents.get('test_agent')
+        # Check API call.
+        assert mock_get.call_args.args[0] == f'{DEFAULT_LOCAL_API_URL}/api/projects/mindsdb/agents/test_agent'
+        expected_agent = Agent(
+            'test_agent',
+            'test_model',
+            [],
+            {},
+            created_at,
+            updated_at
+        )
+        assert agent == expected_agent
+
+    @patch('requests.Session.post')
+    def test_create(self, mock_post):
+        created_at = dt.datetime(2000, 3, 1, 9, 30)
+        updated_at = dt.datetime(2001, 3, 1, 9, 30)
+        data = {
+            'id': 1,
+            'name': 'test_agent',
+            'project_id': 1,
+            'model_name': 'test_model',
+            'skills': [{
+                'id': 0,
+                'name': 'test_skill',
+                'project_id': 1,
+                'type': 'test',
+                'params': {'k1': 'v1'},
+            }],
+            'params': {'k1': 'v1'},
+            'created_at': created_at,
+            'updated_at': updated_at
+        }
+        response_mock(mock_post, data)
+
+        # Create the agent.
+        server = mindsdb_sdk.connect()
+        new_agent = server.agents.create(
+            name='test_agent',
+            model=Model(None, {'name':'m1'}),
+            skills=[Skill('test_skill', 'test', {'k1': 'v1'})],
```

Comment:
Maybe to forbid to use generic Skill and force the user to use specific skill: SQLSkill or KBSkill ?
It helps to send only required params.

Reply:
For now (when we don't have all wrappers for all skills) we could keep to use just generic Skill class